### ISSUE TITLE
bug: CocoaPods could not find compatible versions for pod "GTMSessionFetcher/Core"

### ISSUE THREAD

**Issue by AE1NS** [capawesome-team/capacitor-firebase#375]
### Plugin(s)

Authentication

### Did you test the latest version?

1.4.0-dev.6ebed57.1680891069

### Platform(s)

iOS

### Current behavior

I just upgraded to capacitor 5 and when I run pod install I get this issue:
```
[!] CocoaPods could not find compatible versions for pod "GTMSessionFetcher/Core":
  In snapshot (Podfile.lock):
    GTMSessionFetcher/Core (< 4.0, = 3.1.1, >= 2.1)

  In Podfile:
    CapacitorFirebaseAuthentication/Google (from `../../node_modules/.pnpm/@capacitor-firebase+authentication@1.4.0-dev.6ebed57.1680891069_@capacitor+core@5.0.0_firebase@9.17.1/node_modules/@capacitor-firebase/authentication`) was resolved to 1.4.0-dev.6ebed57.1680891069, which depends on
      GoogleSignIn (= 6.2.4) was resolved to 6.2.4, which depends on
        GTMSessionFetcher/Core (< 3.0, >= 1.1)
```

### Expected behavior

No errors

### Steps to reproduce

pod install

### Capacitor doctor

```
@capacitor/cli: 5.0.0
@capacitor/core: 5.0.0
@capacitor/android: 5.0.0
@capacitor/ios: 5.0.0
```

**Comment by robingenz** [comment]
Please paste the full error log.

**Comment by AE1NS** [comment]
Sorry, I made a mistake here. This issue only occurs with my script I have to use, to support pnpm. See https://github.com/capawesome-team/capacitor-firebase/discussions/308.

Lets take the authentication implementation as example. I needed the following steps there to get it run (done by the script from the discussion):
- Remove the `CapacitorFirebaseAuthentication/Google` pod entry, that maybe have an invalid path because of pnpm
- Do a `npx cap sync ios` to update the `capacitor_pods` and therefore also the `CapacitorFirebaseAuthentication`, which sets the correct pnpm path to the package.
- Re-add `CapacitorFirebaseAuthentication/Google` and take the base path from `CapacitorFirebaseAuthentication`, thats now correctly set
- Do another `npx cap sync ios` and we are done

So in my case the exception occurs in the last step. Here is a simplified reproduction:
- Do a `npx cap sync ios` which adds `CapacitorFirebaseAuthentication` to the podfile. This runs pod install without any issues.
- Now add the additional pod for authentication `pod 'CapacitorFirebaseAuthentication/Google', :path => '...'` and do a `pod install`

This crashes now with the error above. When I remove the Podfile.lock and do a `pod install` it completes without errors. So when the Podfile is complete from the beginning, it seems to work without issues. Do you have any ideas how to solve this?

I would give you more logs, but I dont know how.

**Comment by robingenz** [comment]
Just paste the whole console output or provide a Minimal, Reproducible Example. I guess you use another Capacitor plugin with which there are dependency conflicts.

**Comment by AE1NS** [comment]
`cd ios/App`
`pod deintegrate`
```
% pod deintegrate
Deintegrating `App.xcodeproj`
Deleted 1 'Copy Pods Resources' build phases.
Deleted 1 'Check Pods Manifest.lock' build phases.
Deleted 1 'Embed Pods Frameworks' build phases.
- Pods_Dev.framework
- Pods-Dev.debug.xcconfig
- Pods-Dev.release.xcconfig
Deleted 1 'Copy Pods Resources' build phases.
Deleted 1 'Check Pods Manifest.lock' build phases.
Deleted 1 'Embed Pods Frameworks' build phases.
- Pods_Portazon.framework
- Pods-Portazon.debug.xcconfig
- Pods-Portazon.release.xcconfig
Deleted 1 'Copy Pods Resources' build phases.
Deleted 1 'Check Pods Manifest.lock' build phases.
Deleted 1 'Embed Pods Frameworks' build phases.
- Pods_PortazonStore.framework
- Pods-PortazonStore.debug.xcconfig
- Pods-PortazonStore.release.xcconfig
Deleted 1 'Check Pods Manifest.lock' build phases.
- Pods_NSE_Dev.framework
- Pods-NSE_Dev.debug.xcconfig
- Pods-NSE_Dev.release.xcconfig
Deleted 1 'Check Pods Manifest.lock' build phases.
- Pods_NSE_Portazon.framework
- Pods-NSE_Portazon.debug.xcconfig
- Pods-NSE_Portazon.release.xcconfig
Deleted 1 empty `Pods` groups from project.
Removing `Pods` directory.

Project has been deintegrated. No traces of CocoaPods left in project.
Note: The workspace referencing the Pods project still remains.
```
`pod install` (without `CapacitorFirebaseAuthentication/Google` pod)
```
% pod install
Analyzing dependencies
Downloading dependencies
Installing AeinsCapacitorHttpWifiBinded (5.0.1)
Installing AeinsCapacitorLocationState (5.0.0)
Installing AeinsCapacitorStartMaps (5.0.0)
Installing AeinsGoogleMaps (5.0.1-alpha-0.5)
Installing Airship (16.9.3)
Installing AirshipGimbalAdapter (4.1.2)
Installing AirshipServiceExtension (16.11.3)
Installing AparajitaCapacitorBiometricAuth (4.0.0)
Installing AparajitaCapacitorDarkMode (3.3.3)
Installing AparajitaCapacitorSecureStorage (3.0.1)
Installing Capacitor (5.0.0)
Installing CapacitorApp (5.0.0)
Installing CapacitorBlobWriter (0.0.1)
Installing CapacitorBrowser (5.0.0)
Installing CapacitorClipboard (5.0.0)
Installing CapacitorCommunityBarcodeScanner (3.0.3)
Installing CapacitorCommunityFileOpener (1.0.5)
Installing CapacitorCommunityKeepAwake (3.0.0)
Installing CapacitorCommunityScreenBrightness (5.0.0)
Installing CapacitorCordova (5.0.0)
Installing CapacitorDevice (5.0.0)
Installing CapacitorFilesystem (5.0.0)
Installing CapacitorFirebaseAnalytics (1.4.0-dev.6ebed57.1680891069)
Installing CapacitorFirebaseAuthentication (1.4.0-dev.6ebed57.1680891069)
Installing CapacitorFirebaseCrashlytics (1.4.0-dev.6ebed57.1680891069)
Installing CapacitorGeolocation (5.0.0)
Installing CapacitorKeyboard (5.0.0)
Installing CapacitorLocalNotifications (5.0.0)
Installing CapacitorNativeSettings (4.0.3)
Installing CapacitorNetwork (5.0.0)
Installing CapacitorPreferences (5.0.0)
Installing CapacitorRateApp (3.0.0)
Installing CapacitorSplashScreen (5.0.0)
Installing CapacitorStatusBar (5.0.0)
Installing CordovaPlugins (5.0.0)
Installing CordovaPluginsStatic (5.0.0)
Installing FalconetaCapacitorWifiConnect (0.0.33)
Installing Firebase (10.2.0)
Installing FirebaseAuth (10.2.0)
Installing FirebaseCore (10.2.0)
Installing FirebaseCoreInternal (10.9.0)
Installing FirebaseCrashlytics (10.2.0)
Installing FirebaseInstallations (10.9.0)
Installing FreshaCapacitorPluginApplepay (4.0.0)
Installing GCDWebServer (3.5.4)
Installing GTMSessionFetcher (3.1.1)
Installing Gimbal (2.85)
Installing Google-Maps-iOS-Utils (4.1.0)
Installing GoogleDataTransport (9.2.2)
Installing GoogleMaps (7.4.0)
Installing GoogleUtilities (7.11.1)
Installing PromisesObjC (2.2.0)
Installing SparkfabrikCapacitorPluginIdfa (4.0.3)
Installing nanopb (2.30909.0)
Generating Pods project
Integrating client project
Pod installation complete! There are 37 dependencies from the Podfile and 54 total pods installed.
```
Now add the `CapacitorFirebaseAuthentication/Google` pod
`pod install`
```
% pod install
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "GTMSessionFetcher/Core":
  In snapshot (Podfile.lock):
    GTMSessionFetcher/Core (< 4.0, = 3.1.1, >= 2.1)

  In Podfile:
    CapacitorFirebaseAuthentication/Google (from `../../node_modules/.pnpm/@capacitor-firebase+authentication@1.4.0-dev.6ebed57.1680891069_@capacitor+core@5.0.0_firebase@9.17.1/node_modules/@capacitor-firebase/authentication`) was resolved to 1.4.0-dev.6ebed57.1680891069, which depends on
      GoogleSignIn (= 6.2.4) was resolved to 6.2.4, which depends on
        GTMSessionFetcher/Core (< 3.0, >= 1.1)


You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * changed the constraints of dependency `GTMSessionFetcher/Core` inside your development pod `CapacitorFirebaseAuthentication/Google`.
   You should run `pod update GTMSessionFetcher/Core` to apply changes you've made.
```

I will create an empty capacitor app and try it out there.

**Comment by robingenz** [comment]
Can you share your npm dependencies (package.json)?

**Comment by AE1NS** [comment]
Let me create a fresh reproduction so I will see if its caused by the plugin or any other dependency. I will update you when its done.

**Comment by AE1NS** [comment]
I couldnt reproduce it in a new project, so I compared the versions of the pods that got installed. I just saw that my new project used the 5.0.0 version of your plugin that got just released. My other project with the issue used 1.4.0-dev... so I updated your plugins to 5.0.0 and now the issue is gone.

Thank you for your fast responses! I really appreciate your work you put in this pool of firebase plugins! :)
